analysis_options.yaml

```diff
@@ -153,6 +155,7 @@ linter:
     # - type_annotate_public_apis # subset of always_specify_types
     - type_init_formals
     # - unawaited_futures # too many false positives
+    # - unnecessary_await_in_return # not yet tested
```

Comment:
i don't really like this one. it leads to weird code where half the returns return `X` and the other half return `Future<X>`, which is confusing. I'd much rather have the opposite, which is that if you're `async`, you have to return non-Futures, and if you're not, you have to return the exact type.

Reply:
I think it would make sense for the compiler to optimize away things like "return await", certainly. But that's something the compiler can take care of.

Reply:
I don't think compiler can optimize `return await xxx` to be handled like `return xxx` because doing that will change the semantic of code.

Every `await` schedules a microtask (I see several codes using `await null` to release the thread an let another task to do things) and it would be misleading if `await` have a different behaviour depending on the presence of a prepending `return`.